lib/config/sfApplicationConfiguration.class.php

```diff
@@ -143,7 +143,8 @@ public function initConfiguration()
     }
 
     // error settings
-    ini_set('display_errors', $this->isDebug() ? 'on' : 'off');
+    // PHP >= 5.2.4 supports stdout/stderr instead of bool here: https://www.php.net/manual/en/errorfunc.configuration.php#ini.display-errors
+    ini_set('display_errors', $this->isDebug() ? sfConfig::get('sf_display_errors', 'stdout') : '');
```

Comment:
To fully tackle this part: the correct behaviour is found by newer Symfony version (2+) and become this:

```diff
-        ini_set('display_errors', $this->isDebug() ? 'on' : 'off');
+        if (null === $displayErrors = sfConfig::get('sf_display_errors')) {
+            $displayErrors = true; // Even when debug is off we want to see errors on output/console.
+        }
+        if ('cli' !== PHP_SAPI) {
+            ini_set('display_errors', 0);
+        } elseif ($displayErrors && (!ini_get('log_errors') || ini_get('error_log'))) {
+            // CLI - display errors only if they're not already logged to STDERR
+            ini_set('display_errors', 1);
+        }
```

This patch done this:

> Except for CLI - display errors only if they're not already logged to STDERR

See https://github.com/symfony/symfony/blob/5.1/src/Symfony/Component/ErrorHandler/Debug.php#L25-L30

Reply:
Thanks, that would mean, display_errors is always off, if not run in cli. Should we also consider phpdbg as shown in the diff you referred to?

Reply:
And maybe we should execute the whole thing only if $this->isDebug() is on, right?

Reply:
Web context
-------

display_errors should always be off.
This cause issue on HTML, json, ... format rendering.

Console
-----

Like said the comment:
> display errors only if they're not already logged to STDERR

Problematic
--------

Fatal error are not handled by sf1 error handler, there are 2 solutions:

1. Use symfony/debug and disable error handling by sfWebDebugLogger
2. enable PHP ini `log_errors` to log the error on server logs.

---

What problematic do you try to solve? can you add more details on PR description?

---

For sure le latest version of the logic is on symfony/symfony repository.
The only effort is to make it compatible with PHP 5.3 =.> PHP 7.4